src/app/src/util/util.js

```diff
@@ -633,3 +634,35 @@ export const pluralizeResultsCount = (count) => {
 
     return `${count} results`;
 };
+
+export const removeDuplicatesFromOtherLocationsData = otherLocationsData => uniqWith(
+    otherLocationsData,
+    (location, otherLocation) => {
+        const lat = get(location, 'lat', null);
+        const lng = get(location, 'lng', null);
+        const name = get(location, 'contributor_name', null);
+        const id = get(location, 'contributor_id', null);
+
+        const otherLat = get(otherLocation, 'lat', null);
+        const otherLng = get(otherLocation, 'lng', null);
+        const otherName = get(otherLocation, 'contributor_name', null);
+        const otherID = get(otherLocation, 'contributor_id', null);
+
+        if (lat !== otherLat || lng !== otherLng) {
+            return false;
+        }
+
+        if ((!id && otherID) ||
+            (!name && otherName) ||
+            (id && !otherID) ||
+            (name && !otherName)) {
+            return true;
+        }
+
+        if (name === otherName || id === otherID) {
```

Comment:
Do we need to also be checking the contributor name if we are checking IDs? There should be no case where the IDs match and the names do not.